### PR TITLE
Feature/lp113-office-viewmodel

### PR DESCRIPTION
**Pull request by ymorenoanz**
I've created the OfficeViewModel

### CODE REVIEW

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 1]
```diff
@@ -29,43 +31,55 @@ import com.iteneum.office.R
  * Comment
  */
 @Composable
-fun OfficeUI() {
-
+fun OfficeUI(vm: OfficeViewModel = viewModel()) //Injecting the viewModel
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Please be more descriptive in the name of the variable vm.

Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: If you are going to define the function onClick in the viewModel, please invoke the function in the UI

File: feature/office/src/main/java/com/iteneum/office/presentation/viewmodel/OfficeViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import javax.inject.Inject
+
+/**
+ * This class is the viewModel corresponding to the Office screen
+ * @author Yaritza Moreno
+ */
+@HiltViewModel
+class OfficeViewModel @Inject constructor(): ViewModel() {
+
+    private fun onClick(){
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/viewmodel/OfficeViewModel.kt]: This function shouldn't be private.
Please be more descriptive on which on click is executed

**Comment by melissaquezada94** [comment]
If the task is not related to the UI, you don't need to add screenshot of the code because I will be able to watch it on the files changed section

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt (hunk 1 above)
Comment (carlosph816) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: You need to change the viewModel() for hiltViewModel() cause we will use navigation compose and viewModel() doesn't work on it

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 3]
```diff
@@ -1,5 +1,7 @@
-package com.iteneum.office.presentation
+package com.iteneum.office.presentation.ui
 
+import android.content.Context
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Would you please remove the unused imports and do the reformat code?

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 4]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
 import com.iteneum.designsystem.components.LpOutlinedButton
 import com.iteneum.designsystem.theme.LeasePertTheme
 import com.iteneum.office.R
+import com.iteneum.office.presentation.viewmodel.OfficeViewModel
 
 /**
  * This function creates the Office screen UI
  *With the Outlined buttons and the company information
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Please add a space in between this: `*With`

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 5]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
+ * @param vm injects the viewModel into the UI
  * @author Yaritza Moreno
  * Comment
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Please delete the word `Comment`

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 6]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
  */
 @Composable
-fun OfficeUI() {
-
+fun OfficeUI(vm: OfficeViewModel = hiltViewModel())
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Please be more descriptive declaring the variables name in this case viewModel

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 7]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
+{
     Column() {
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: You can add the property `fillMaxSize` to this Column

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 8]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
 
         //Get sizes from LeasePertTheme archive
         val sizes = LeasePertTheme.sizes
 
-        Text(text = stringResource(id = R.string.LPInfo),
+        val context: Context
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: If you are not using this variable context, please delete it

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 9]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
+
+        Text(
+            text = stringResource(id = R.string.LPInfo),
             modifier = Modifier
                 .width(sizes.extraSize124)
                 .height(sizes.regularSize),
-            style = TextStyle(textAlign = TextAlign.Justify, color = Color.Gray,
+            style = TextStyle(
+                textAlign = TextAlign.Justify, color = Color.Gray,
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: Would you add the property color to the next line to have a good structure in the code, please?

File: feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt [hunk 10]
```diff
@@ -18,54 +20,72 @@ import androidx.compose.ui.text.style.TextAlign
...
             icon = Icons.Filled.Call,
             textButton = stringResource(id = R.string.LPCallButton),
-            onClick = { /*TODO*/ } )
+            onClick = { vm.onClick() })
 
         LpOutlinedButton(modifier = Modifier,
             icon = Icons.Outlined.Mail,
             textButton = stringResource(id = R.string.LPMailButton),
-            onClick = { /*TODO*/ } )
+            onClick = { vm.onClick() })
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/ui/Office.kt]: You can add a new onClick for this function called: onMailClicked()

File: feature/office/src/main/java/com/iteneum/office/presentation/viewmodel/OfficeViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * This class is the viewModel corresponding to the Office screen
+ * @author Yaritza Moreno
+ */
+@HiltViewModel
+class OfficeViewModel @Inject constructor(): ViewModel() {
+
+    //I'm just added the context parameter to test the viewModel
+    //It can be removed in the future
+    fun onClick(){
```
Comment (melissaquezada94) [inline feature/office/src/main/java/com/iteneum/office/presentation/viewmodel/OfficeViewModel.kt]: Please be more descriptive on what is going to be clicked on the function. For example:

1. When the user does click on the `Call` button you can name the action like: onCallClicked()
2. When the user does click ib the `Mail` button you can name the action like: onMailClicked()

Also you can add the two functions for each bottom, remove the viewModelScope and add a TODO comment like:
//TODO add the functionality for the call button 
//TODO add the functionality for the mail button

**Review by melissaquezada94 — APPROVED** [review]

**Review by carlosph816 — APPROVED** [review]
Well done

**Pull request merged**